clazy is complaining:
```
Error: /home/runner/work/mixxx/mixxx/src/effects/effectsmanager.cpp:177:10: error: Missing reference in range-for with non trivial type (QSharedPointer<EqualizerEffectChain>) [-Wclazy-range-loop-reference]
    for (auto pEqChainSlot : std::as_const(m_equalizerEffectChains)) {
         ^
         const  &
1 error generated.
```
The rest looks good. Thank you. Please confirm when it is ready for merge.

--- a/src/effects/effectsmanager.cpp
+++ b/src/effects/effectsmanager.cpp
@@ -174,9 +174,7 @@ void EffectsManager::addQuickEffectChain(const ChannelHandleAndGroup& deckHandle
 
 void EffectsManager::loadDefaultEqsAndQuickEffects() {
     auto pDefaultEqEffect = m_pChainPresetManager->getDefaultEqEffect();
-    QHashIterator<QString, EqualizerEffectChainPointer> eqIt(m_equalizerEffectChains);
-    while (eqIt.hasNext()) {
-        auto pEqChainSlot = eqIt.next().value();
+    for (const auto& pEqChainSlot : std::as_const(m_equalizerEffectChains)) {
         const auto pEqEffectSlot = pEqChainSlot->getEffectSlot(0);
         VERIFY_OR_DEBUG_ASSERT(pEqEffectSlot) {
             return;
@@ -186,9 +184,7 @@ void EffectsManager::loadDefaultEqsAndQuickEffects() {
 
     const auto pDefaultQuickEffectPreset =
             m_pChainPresetManager->getDefaultQuickEffectPreset();
-    QHashIterator<QString, QuickEffectChainPointer> qeIt(m_quickEffectChains);
-    while (qeIt.hasNext()) {
-        auto pChainSlot = qeIt.next().value();
+    for (const auto& pChainSlot : std::as_const(m_quickEffectChains)) {
         pChainSlot->loadChainPreset(pDefaultQuickEffectPreset);
     }
 }
@@ -214,11 +210,7 @@ void EffectsManager::readEffectsXml() {
 
     // Note: QuickEffect and EQ chains are created only for existing main decks,
     // thus only for those the configured presets are requested
-    QStringList deckStrings;
-    QHashIterator<QString, QuickEffectChainPointer> qeIt(m_quickEffectChains);
-    while (qeIt.hasNext()) {
-        deckStrings << qeIt.next().key();
-    }
+    const QStringList deckStrings = m_quickEffectChains.keys();
     EffectsXmlData data = m_pChainPresetManager->readEffectsXml(doc, deckStrings);
 
     for (int i = 0; i < data.standardEffectChainPresets.size(); i++) {

--- a/src/effects/effectsmanager.cpp
+++ b/src/effects/effectsmanager.cpp
@@ -298,6 +298,7 @@ void EffectsManager::saveEffectsXml() {
     }
 
     QHash<QString, EffectChainPresetPointer> quickEffectChainPresets;
+    quickEffectChainPresets.reserve(m_quickEffectChains.size());
     QHashIterator<QString, QuickEffectChainPointer> qeIt(m_quickEffectChains);
     while (qeIt.hasNext()) {
         qeIt.next();
@@ -306,6 +307,7 @@ void EffectsManager::saveEffectsXml() {
     }
 
     QList<EffectChainPresetPointer> standardEffectChainPresets;
+    standardEffectChainPresets.reserve(m_quickEffectChains.size());
     for (const auto& pChainSlot : std::as_const(m_standardEffectChains)) {
         auto pPreset = EffectChainPresetPointer::create(pChainSlot.data());
         standardEffectChainPresets.append(pPreset);

--- a/src/effects/effectsmanager.cpp
+++ b/src/effects/effectsmanager.cpp
@@ -302,14 +302,16 @@ void EffectsManager::saveEffectsXml() {
     QHashIterator<QString, QuickEffectChainPointer> qeIt(m_quickEffectChains);
     while (qeIt.hasNext()) {
         qeIt.next();
-        auto pPreset = EffectChainPresetPointer::create(qeIt.value().data());
+        auto* pQuickEffectChain = qeIt.value().data();
+        auto pPreset = EffectChainPresetPointer::create(pQuickEffectChain);
         quickEffectChainPresets.insert(qeIt.key(), pPreset);
     }
 
     QList<EffectChainPresetPointer> standardEffectChainPresets;
     standardEffectChainPresets.reserve(m_quickEffectChains.size());
     for (const auto& pChainSlot : std::as_const(m_standardEffectChains)) {
-        auto pPreset = EffectChainPresetPointer::create(pChainSlot.data());
+        auto* pChain = pChainSlot.data();
+        auto pPreset = EffectChainPresetPointer::create(pChain);
         standardEffectChainPresets.append(pPreset);
     }
 

